### PR TITLE
Changelog for rc.3

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,28 @@
...
 
 **The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
 
+## Features ✨ and improvements 🏁
+* Introduce static MapboxMap.clearData(resourceOptions: ResourceOptions, callback: AsyncOperationResultCallback) API and MapboxMap#clearData(callback: AsyncOperationResultCallback), Snapshotter#clearData(callback: AsyncOperationResultCallback) APIs. ([#442](https://github.com/mapbox/mapbox-maps-android/pull/442))
+* Optimise the Style#getLayer and Style#getSource APIs' performance. ([#449](https://github.com/mapbox/mapbox-maps-android/pull/449))
+* MapEvents#MAP_LOADING_ERROR events now include source and tile information where appropriate. New fields would allow developers to understand what source or tile has failed to load and the reason for a failure.
+
+## Bug fixes 🐞
+* Fix annotation update missing issue. ([#441](https://github.com/mapbox/mapbox-maps-android/pull/441))
```
Comment (tobrun) [inline CHANGELOG.md:15]: This could be explained a bit more in detail. When reading this I'm not getting why it was an issue ad what the fix was.

**Review by tobrun — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,28 @@
...
+* Perform annotation click synchronously and change AnnotationManagerImpl#queryMapForFeatures function to be synchronous. ([#455](https://github.com/mapbox/mapbox-maps-android/pull/455))
+* Fix raster/v1 terrain tiles fetch failures caused by appending pixel ratio to the URLs when tile size is equal to 512.
+* Improve persistent layer pinning by keeping information about initial LayerPosition.
+* Disable MapboxTelemetryInitProvider if the telemetry is disabled via app's manifest.
```
Comment (pengdev) [inline CHANGELOG.md:19]: maybe describe this might affect the map's initialisation time?

Comment (pengdev) [reply CHANGELOG.md:19]: And link to the ([#457](https://github.com/mapbox/mapbox-maps-android/pull/457))

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,28 @@
...
+* Introduce static MapboxMap.clearData(resourceOptions: ResourceOptions, callback: AsyncOperationResultCallback) API and MapboxMap#clearData(callback: AsyncOperationResultCallback), Snapshotter#clearData(callback: AsyncOperationResultCallback) APIs. ([#442](https://github.com/mapbox/mapbox-maps-android/pull/442))
+* Optimise the Style#getLayer and Style#getSource APIs' performance. ([#449](https://github.com/mapbox/mapbox-maps-android/pull/449))
+* MapEvents#MAP_LOADING_ERROR events now include source and tile information where appropriate. New fields would allow developers to understand what source or tile has failed to load and the reason for a failure.
+
+## Bug fixes 🐞
+* Fix annotation update missing issue. ([#441](https://github.com/mapbox/mapbox-maps-android/pull/441))
+* Perform annotation click synchronously and change AnnotationManagerImpl#queryMapForFeatures function to be synchronous. ([#455](https://github.com/mapbox/mapbox-maps-android/pull/455))
+* Fix raster/v1 terrain tiles fetch failures caused by appending pixel ratio to the URLs when tile size is equal to 512.
+* Improve persistent layer pinning by keeping information about initial LayerPosition.
```
Comment (pengdev) [inline CHANGELOG.md:18]: Fixed an issue that the LayerPosition is not persisted across the style change, when using persistent layer based annotation plugin and location component plugin.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,10 +2,28 @@
...
 **The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
 
+## Features ✨ and improvements 🏁
+* Introduce static MapboxMap.clearData(resourceOptions: ResourceOptions, callback: AsyncOperationResultCallback) API and MapboxMap#clearData(callback: AsyncOperationResultCallback), Snapshotter#clearData(callback: AsyncOperationResultCallback) APIs. ([#442](https://github.com/mapbox/mapbox-maps-android/pull/442))
+* Optimise the Style#getLayer and Style#getSource APIs' performance. ([#449](https://github.com/mapbox/mapbox-maps-android/pull/449))
+* MapEvents#MAP_LOADING_ERROR events now include source and tile information where appropriate. New fields would allow developers to understand what source or tile has failed to load and the reason for a failure.
+
+## Bug fixes 🐞
+* Fix annotation update missing issue. ([#441](https://github.com/mapbox/mapbox-maps-android/pull/441))
+* Perform annotation click synchronously and change AnnotationManagerImpl#queryMapForFeatures function to be synchronous. ([#455](https://github.com/mapbox/mapbox-maps-android/pull/455))
```
Comment (pengdev) [inline CHANGELOG.md:16]: Is this a breaking change?

File: CHANGELOG.md (hunk 3 above)
Comment (pengdev) [reply CHANGELOG.md:18]: And link to the gl-native bump PR?

File: CHANGELOG.md (hunk 4 above)
Comment (kiryldz) [reply CHANGELOG.md:16]: More yes than no, thanks for reminding! Script did not put it to breaking change section for some reason.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,10 +2,30 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
-# 10.0.0-rc.2 June 23, 2021
+# 10.0.0-rc.3 June 30, 2021
 
 **The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
 
+## Breaking changes ⚠️
+* Perform annotation click synchronously and change AnnotationManagerImpl#queryMapForFeatures function to be synchronous. ([#455](https://github.com/mapbox/mapbox-maps-android/pull/455))
```
Comment (pengdev) [inline CHANGELOG.md:10]: maybe also include the API surface that's changed?

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,10 +2,30 @@
...
+* Optimise the Style#getLayer and Style#getSource APIs' performance. ([#449](https://github.com/mapbox/mapbox-maps-android/pull/449))
+* MapEvents#MAP_LOADING_ERROR events now include source and tile information where appropriate. New fields would allow developers to understand what source or tile has failed to load and the reason for a failure. ([#457](https://github.com/mapbox/mapbox-maps-android/pull/457))
+
+## Bug fixes 🐞
+* Fix dropping annotation source updates if those were emitted rapidly without handler. ([#441](https://github.com/mapbox/mapbox-maps-android/pull/441))
+* Fix raster/v1 terrain tiles fetch failures caused by appending pixel ratio to the URLs when tile size is equal to 512. ([#457](https://github.com/mapbox/mapbox-maps-android/pull/457))
+* Fixed an issue that the LayerPosition is not persisted across the style change, when using persistent layer based annotation plugin and location component plugin. ([#457](https://github.com/mapbox/mapbox-maps-android/pull/457))
+* Disable MapboxTelemetryInitProvider if the telemetry is disabled via app's manifest reducing startup time. ([#442](https://github.com/mapbox/mapbox-maps-android/pull/442))
```
Comment (pengdev) [inline CHANGELOG.md:21]:
```suggestion
* Disable MapboxTelemetryInitProvider if the telemetry is disabled via app's manifest, and reduces startup time. ([#442](https://github.com/mapbox/mapbox-maps-android/pull/442))
```

**Review by pengdev — APPROVED** [review]
LGTM % nits

File: CHANGELOG.md (hunk 5 above)
Comment (kiryldz) [reply CHANGELOG.md:10]: I will do that in release notes.

File: CHANGELOG.md (hunk 6 above)
Comment (kiryldz) [reply CHANGELOG.md:21]: I'm an enemy of commas so I'll leave it my way if you don't mind 😉

**Pull request merged**